homework/index.js

```diff
@@ -1,46 +1,94 @@
+  Object.keys(options).forEach(key => {
+    const value = options[key];
+    if (key === 'text') {
+      elem.textContent = value;
+    } else {
+      elem.setAttribute(key, value);
+    }
+  });
+  return elem;
+}
+
+async function getContributorInformation(data) {
+  try {
+    const contribs = await fetch(contribsUrl);
+    const jsonData = await contribs.json();
+    jsonData.forEach(contributor => {
+      createAndAppend('div', contribs, { text: contributor.login, class: 'contributor' })
+      createAndAppend('img', contribs, { src: contributor.avatar_url, height: 150, widtth: 150, id: 'img' })
+      createAndAppend('div', contribs, { text: contributor.contributions })
+    })
+  } catch (err) {
+    const root = document.getElementById('root');
+    createAndAppend('div', contribs, { text: err.message, class: 'alert-error' })
   }
+}
+
+function addInfo(label, value) {
+  const container = createAndAppend('div', repoInfo, { class: 'container' });
+  createAndAppend('span', container, { text: label });
+  createAndAppend('span', container, { text: value });
+};
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
-      }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
-    });
+async function main(url) {
+  try {
+    const response = await fetch(url);
+    const json = await response.json();
+    const root = document.getElementById('root');
+    const select = createAndAppend('select', root, { class: 'select' });
+    createAndAppend('option', select, { text: 'Choose your favorite repo' });
+    let sorted = json.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase()));
+    sorted.forEach(repo => {
+      const name = repo.name;
+      createAndAppend('option', select, { text: name });
+      const wraper = createAndAppend('div', root, { class: 'wraper' });
+      const repoInfo = createAndAppend('div', wraper, { class: 'repoinfo' });
+      const contribs = createAndAppend('div', wraper, { class: 'contribs' });
+      select.addEventListener('change', evt => {
+        const selectedRepo = evt.target.value;
+        const repo = json.filter(r => r.name == selectedRepo)[0];
+        repoInfo.innerHTML = '';
+        contribs.innerHTML = '';
+        addInfo('Name: ', repo.name);
+        addInfo('Desciption: ', repo.description);
+        addInfo('Number of forks: ', repo.forks);
+        addInfo('Updated: ', repo.updated_at)
+        const contribsUrl = repo.contributors_url;
+        getContributorInformation(contribsUrl, contribs)
+      })
+
+
+    })
+  }
+  catch (err) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, { text: err.message, class: 'alert-error' })
   }
 
+
   const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
```

Comment:
It looks like you call the main function from inside of the main function, move this line and the window.onload outside of the `}` on the last row in the file

Reply:
Fixed that! Thanks.

Reply:
I made these changes. Everything looks fine except for the contributors. I changed the placement of "getContributorInformation" function lots of times but couldn't get it to work.